kafka_quota_enforcer/src/main/java/com/tesla/data/quota/enforcer/ConfiguredQuota.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2021. Tesla Motors, Inc. All rights reserved.
+ */
+
+package com.tesla.data.quota.enforcer;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import tesla.shade.com.google.common.base.MoreObjects;
+import tesla.shade.com.google.common.base.Objects;
+import tesla.shade.com.google.common.base.Preconditions;
+
+/**
+ * ConfiguredQuota represents a quota and its desired configuration.
+ */
+public class ConfiguredQuota {
+  private final String principal;
+  private final String client;
+  @JsonProperty("producer-byte-rate")
+  private final Integer producerByteRate;
+  @JsonProperty("consumer-byte-rate")
+  private final Integer consumerByteRate;
+  @JsonProperty("request-percentage")
+  private final Integer requestPercentage;
+
+  public ConfiguredQuota(String principal, String client, Integer producerByteRate,
+                         Integer consumerByteRate, Integer requestPercentage) {
+    Preconditions.checkArgument(principal != null || client != null, "Invalid quota configuration detected. " +
+        "At least one of 'principal' or 'client' must be provided.");
+    this.principal = principal;
+    this.client = client;
+    this.producerByteRate = producerByteRate;
+    this.consumerByteRate = consumerByteRate;
+    this.requestPercentage = requestPercentage;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
minor style nit: generally equals/hashcode/toString are at the bottom of the class

Reply:
generally hashcode goes next to equals - they should be logically the same